### PR TITLE
Moment units redux

### PR DESCRIPTION
**Pull request by rosteen**
Updates the recent Moment Maps unit work to better align with PO specs by displaying the actual physical units (including flux units for moment 0), and giving the option to reduce all higher moments to velocity rather than velocity^N.

### CODE REVIEW

**Comment by kecnry** [comment]
Looks like tests need updating for the new reduced units.  The `TypeError: __init__() got an unexpected keyword argument 'out'` error is unrelated though and caused by the recent release of glue/glue-jupyter.

**Comment by rosteen** [comment]
I'm also still debugging something I found after opening the PR, I'll fix tests once I'm done that.

**Comment by rosteen** [comment]
Here's a quick demo of what this looks like now:


https://github.com/spacetelescope/jdaviz/assets/39831871/ae9d2c89-ab1c-44f6-88c7-8348813b8ea6

File: jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py
Comment (kecnry) [inline jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py:155]: how does this look to the user using the API?

Comment (rosteen) [reply jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py:155]: The user still sees the full options list in the API.

Comment (kecnry) [reply jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py:155]: If anything isn't valid, we should raise errors in `calculate_moment` then?

File: jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py
Comment (bmorris3) [inline jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py:125]: When using the UI, if you delete the number in the moment field, you get the following traceback:
```
File ~/git/jdaviz/jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py:126, in MomentMap._set_data_units(self, event)
    123 @observe("dataset_selected", "n_moment")
    124 def _set_data_units(self, event={}):
--> 126     unit_options_index = 2 if self.n_moment > 2 else self.n_moment
    127     if self.output_unit_selected not in moment_unit_options[unit_options_index]:
    128         self.output_unit_selected = moment_unit_options[unit_options_index][0]

TypeError: '>' not supported between instances of 'str' and 'int'
```
because `self.n_moment` is an empty string. Here's a possible fix?


```suggestion
        if isinstance(self.n_moment, str): 
            return
```

**Review by bmorris3 — CHANGES_REQUESTED** [review]
One fix below.

File: jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py
Comment (kecnry) [inline jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py:185]:
```suggestion
                             f"{moment_unit_options[unit_options_index]} for "
```

File: jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.vue
Comment (kecnry) [inline jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.vue:87]: I can get this same error message by setting n_moment to 1 and output_unit to Flux... granted that is only possible by the UI, so I'm not sure its worth an extra case here to show the correct error message or not.

**Review by kecnry — COMMENTED** [review]
This seems quite intuitive from the UI when changing moment (it remembers the user selection when it still exists, but defaults to a new value when it doesn't).


https://github.com/spacetelescope/jdaviz/assets/877591/43bc5a62-031e-481c-a46a-26cf7f24776a



From the API, there are still a few weird cases, but those handled by the error checking in `calculate_moment`, which should be sufficient.

File: jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py
Comment (kecnry) [inline jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py:125]:
```suggestion
        if isinstance(self.n_moment, str):
```

File: jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py
Comment (rosteen) [reply jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py:89]: Looks like you're right, I'll remove this.

Comment (rosteen) [inline jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py:89]:
```suggestion
```

**Comment by rosteen** [comment]
> One comment remaining above, but I won't insist if it isn't easy! Thanks!

I added a separate case in the vue template for this, good suggestion to make this more accurate.

**Review by bmorris3 — APPROVED** [review]
Approving so that my prior requested-changes do not block merge.

**Pull request merged**
